### PR TITLE
youki: init at 0.0.4

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/youki/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  src = fetchFromGitHub {
+    owner = "containers";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-XwapCfu6Me0xSe+qFz9BFRYpQvG+ztb6QyhGejYRPb4=";
+  };
+
+  buildInputs = [ dbus libseccomp systemd ];
+  nativeBuildInputs = [ pkg-config installShellFiles ];
```
Comment (figsoda) [inline pkgs/applications/virtualization/youki/default.nix:23]: doesn't really matter but usually `nativeBuildInputs` comes before `buildInputs`
```suggestion
  nativeBuildInputs = [ pkg-config installShellFiles ];

  buildInputs = [ dbus libseccomp systemd ];
```

File: pkgs/applications/virtualization/youki/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  postInstall = ''
+    rm $out/bin/integration_test $out/bin/runtimetest
```
Comment (figsoda) [inline pkgs/applications/virtualization/youki/default.nix:26]: we should probably avoid the extra compilation by doing this instead
```nix
  cargoBuildFlags = [ "-p" "youki" ];
  cargoTestFlags = [ "-p" "youki" ];
```

File: pkgs/applications/virtualization/youki/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    installShellCompletion --cmd youki \
+      --bash <($out/bin/youki completion -s bash) \
+      --fish <($out/bin/youki completion -s fish) \
+      --zsh <($out/bin/youki completion -s zsh)
+  '';
+
+  cargoSha256 = "sha256-PT1kVo4gQFH9sIprEoAioNvDL/soMHcA2utEiQJPS/0=";
+
+  doCheck = false;
```
Comment (figsoda) [inline pkgs/applications/virtualization/youki/default.nix:35]: please add a comment on why this is disabled

File: pkgs/applications/virtualization/youki/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  meta = with lib; {
+    description = "A container runtime written in Rust";
+    homepage = "https://containers.github.io/youki/";
```
Comment (figsoda) [inline pkgs/applications/virtualization/youki/default.nix:39]:
```suggestion
    homepage = "https://containers.github.io/youki/";
    changelog = "https://github.com/containers/youki/releases/tag/v${version}";
```

File: pkgs/applications/virtualization/youki/default.nix (hunk 1 above)
Comment (candyc1oud) [reply pkgs/applications/virtualization/youki/default.nix]: I fixed it, thanks

**Comment by figsoda** [comment]
@ofborg eval

**Review by figsoda — APPROVED** [review]

**Pull request merged**